src/app/data/components/data-browser/helper-classes/filter-integration.ts

```diff
@@ -0,0 +1,161 @@
+import { DataBrowserComponent } from "../data-browser.component";
+import { FormArray, FormBuilder, FormGroup } from "@angular/forms";
+import { FilterIntegrationOperator, getAttributeType, getFilterIntegrationOperatorTooltip, getPlaceholderText } from "./search-operators";
+import { getColorForDataType } from "src/app/util/helper-functions";
+
+export class FilterIntegration {
+    private dataBrowser: DataBrowserComponent;
+
+    filterAttributesSSForm: FormGroup;
+    filterAttributesSS: string[];
+    colorsAttributes: string[] = [];
+    uniqueValuesDict: { [key: string]: string[] } = {};
+    operatorsDict: { [key: string]: string[] } = {};
+    tooltipsDict: { [key: string]: string[] } = {};
+
+    constructor(dataBrowser: DataBrowserComponent, private formBuilder: FormBuilder) {
+        this.dataBrowser = dataBrowser;
+        this.formBuilder = formBuilder;
+    }
+
+    initFilterForm() {
+        this.filterAttributesSSForm = this.formBuilder.group({
```

Comment:
What does SS stand for ? :D

Reply:
Similarity Search :) 
in the beginning I thought we would have two forms: one for the similarity search, one for the outliers, but this is not the case anymore, I could remove it from the name so it doesn't make confusion

Reply:
For me both would be okay changing it or keeping it, what is you opinion, is it clear from the context that similarity search is meant?